microsoft/testsuites/hpc/infinibandsuite.py

```diff
@@ -31,6 +33,30 @@
     """,
 )
 class InfinibandSuite(TestSuite):
+    @TestCaseMetadata(
+        description="""
+        This test case will
+        1. List all available network interfaces
+        2. Check if InfiniBand cards are present
+        3. Ensure InfiniBand cards are named starting with "ib0"
+        """,
+        priority=2,
+        requirement=simple_requirement(supported_features=[Infiniband]),
+    )
+    def verify_ib_naming(self, log: Logger, node: Node) -> None:
+        ib_interfaces = node.features[Infiniband].get_ib_interfaces()
+        ib_first_device_name = ib_interfaces[0].ib_device_name
+
+        r = re.compile("ib[0-9]")
+        ib_names = list(filter(r.match, ib_first_device_name))
```

Comment:
`ib_first_device_name` seems like a string. We can simply check using: `r.match(ib_first_device_name)` and it returns none if the regex match fails

Reply:
@darkwhite29 unresolving this. This is still not fixed. `ib_first_device_name` should be a list but is a string instead. You need to do something like this:
```
ib_first_device_name = [ib_interface.ib_device_name for ib_interface in ib_interfaces]
```

Also please update the above naming from `ib_first_device_name -> ib_first_device_names`